cpp/ql/lib/semmle/code/cpp/models/implementations/Iterator.qll

```diff
@@ -428,12 +557,24 @@ class IteratorAssignmentMemberOperator extends MemberFunction {
  * `operator*` and use their own `operator=` to assign to the container.
  */
 private class IteratorAssignmentMemberOperatorModel extends IteratorAssignmentMemberOperator,
-  TaintFunction
+  TaintFunction, SideEffectFunction, AliasFunction
 {
   override predicate hasTaintFlow(FunctionInput input, FunctionOutput output) {
     input.isParameterDeref(0) and
     output.isQualifierObject()
   }
+
+  override predicate hasOnlySpecificReadSideEffects() { any() }
+
+  override predicate hasOnlySpecificWriteSideEffects() { any() }
+
+  override predicate hasSpecificWriteSideEffect(ParameterIndex i, boolean buffer, boolean mustWrite) {
+    i = -1 and buffer = false and mustWrite = false
```

Comment:
Does the same comment apply here that we were discussing above?

Reply:
Oh, yes. I forgot to paste the comment here as well

Reply:
I've added a reference to the other comment in https://github.com/github/codeql/pull/16175/commits/9c9ed13ede5324f6dd37264b0d7e360715e786ee.